pkg/alertmanager/api_grafana.go

```diff
@@ -309,8 +309,25 @@ func (am *MultitenantAlertmanager) SetUserGrafanaConfig(w http.ResponseWriter, r
 		return
 	}
 
-	payload, err := io.ReadAll(r.Body)
+	var input io.Reader
+	maxConfigSize := am.limits.AlertmanagerMaxGrafanaConfigSize(userID)
+	if maxConfigSize > 0 {
+		input = http.MaxBytesReader(w, r.Body, int64(maxConfigSize))
```

Comment:
Is there any practical difference between this and `io.LimitReader` which we use in api.go?

Reply:
I'm happy as-is, maybe we switch the LimitReader usage over in a follow up.